flink-ml-iteration/src/main/java/org/apache/flink/iteration/Iterations.java

```diff
@@ -261,10 +256,6 @@ private static DataStreamList createIteration(
         }
 
         List<DataStream<?>> tailsAndCriteriaTails = new ArrayList<>(tails.getDataStreams());
-        checkState(
```

Comment:
Do we suppose to support termination criteria in unbouned iterations? If we are not doing this, we should probably keep this check.

Reply:
This change is reverted.